File: src/features/quiz/types/question.ts
export type QuestionType = "single_select" | "select_all" | "fill_in_blank";

export interface Question {
  id: string;
  text: string;
  content_type: "movie" | "book" | "both";
  user_age_range: string;
  type: QuestionType;
  /** Placeholder hint for fill_in_blank questions */
  placeholder?: string;
}

File: apps/mobile/lib/core/models/question.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'enums.dart';

part 'question.freezed.dart';
part 'question.g.dart';

/// Quiz question. The anthropic-questions Edge Function returns the subset
/// {id, text, type, options?, placeholder?}; content_type/user_age_range are
/// attached client-side, so they're nullable here.
@freezed
abstract class Question with _$Question {
  const factory Question({
    required String id,
    required String text,
    required QuestionType type,
    @JsonKey(name: 'content_type') ContentType? contentType,
    @JsonKey(name: 'user_age_range') String? userAgeRange,
    List<String>? options,
    String? placeholder,
  }) = _Question;

  factory Question.fromJson(Map<String, dynamic> json) =>
      _$QuestionFromJson(json);
}
